data/scripts.js

```diff
@@ -88,6 +88,28 @@ exports.BattleScripts = {
 		this.useMove(baseMove, pokemon, target, sourceEffect, zMove);
 		this.singleEvent('AfterMove', move, null, pokemon, target, move);
 		this.runEvent('AfterMove', pokemon, target, move);
+
+		// Dancer's activation order is completely different from any other event, so it's handled separately
+		if (move.flags['dance']) {
+			let dancers = [];
+			for (const side of this.sides) {
+				for (const slot of side.active) {
+					let currentPoke = slot;
```

Comment:
Why not just `for (const currentPoke of side.active) {`?

Reply:
Yeah, `slot` is not actually something you want to call a Pokémon.

Reply:
Oh yeah that was from before when I was using for...length and `slot` was `this.battle.sides[i].active[j]`